Lib/blackrenderer/render.py

```diff
@@ -44,10 +46,11 @@ def renderText(
     infos = buf.glyph_infos
     positions = buf.glyph_positions
     glyphLine = buildGlyphLine(infos, positions, glyphNames)
-    bounds = calcGlyphLineBounds(glyphLine, font)
+    bounds = calcGlyphLineBounds(glyphLine, font, settings.useFontMetrics)
     bounds = scaleRect(bounds, scaleFactor, scaleFactor)
-    bounds = insetRect(bounds, -margin, -margin)
-    bounds = intRect(bounds)
+    bounds = insetRect(bounds, -settings.margin, -settings.margin)
+    if not settings.floatBbox:
+        bounds = intRect(bounds)
```

Comment:
We need to figure out how the other backends respond to a float bbox: we may get TypeErrors or undefined behavior. Either way, floats make no sense when you're allocating a bitmap, so perhaps this argument should only be valid in an SVG scope.